vulkan/vulkan_swapchain.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_VULKAN_VULKAN_SWAPCHAIN_H_
+#define FLUTTER_VULKAN_VULKAN_SWAPCHAIN_H_
+
+#include <memory>
+#include <vector>
+#include "flutter/vulkan/vulkan_handle.h"
+#include "lib/ftl/macros.h"
+#include "third_party/skia/include/core/SkSize.h"
+#include "third_party/skia/include/core/SkSurface.h"
+
+namespace vulkan {
+
+class VulkanProcTable;
+class VulkanDevice;
+class VulkanSurface;
+class VulkanBackbuffer;
+class VulkanImage;
+
+class VulkanSwapchain {
+ public:
+  VulkanSwapchain(const VulkanProcTable& vk,
+                  const VulkanDevice& device,
+                  const VulkanSurface& surface,
+                  GrContext* skia_context,
+                  std::unique_ptr<VulkanSwapchain> old_swapchain,
+                  uint32_t queue_family_index);
+
+  ~VulkanSwapchain();
+
+  bool IsValid() const;
+
+  enum class AcquireType {
+    Success,
+    ErrorSurfaceLost,
+    ErrorSurfaceOutOfDate,
+  };
+  using AcquireResult = std::pair<AcquireType, sk_sp<SkSurface>>;
+
+  AcquireResult AcquireSurface();
+
+  FTL_WARN_UNUSED_RESULT
+  bool Submit();
```

Comment:
This behavior is implicitly enforced by the pipelining in the Flutter shell. But yes, since everything in `vulkan/` is agnostic of the Flutter shell, it should likely be documented. Fixed.